# Notebook 01 — Análisis Exploratorio de Datos (EDA)
**Curso:** BD-143 Programación II — CUC 2026
**Dataset:** TMDB Movies 2020–2025


## Carga de datos
Se cargó el dataset TMDB 2020-2025 con 9,999 películas y 15 columnas.
Se identificaron nulos principalmente en backdrop_path (18.81%) y overview (2.78%).

In [1]:
import sys
import os
sys.path.insert(0, os.path.abspath("../src"))

from eda.cargador_datos import CargadorDatos
from eda.procesador_eda import ProcesadorEda

ruta = "../data/raw/tmdb_2020_to_2025.csv"
cargador = CargadorDatos(ruta)
df = cargador.cargar()

Filas cargadas 9999
Valores nulos por columna:
                   cantidad  porcentaje %
adult                     0          0.00
backdrop_path          1881         18.81
genre_ids                 0          0.00
id                        0          0.00
original_language         0          0.00
original_title            0          0.00
overview                278          2.78
popularity                0          0.00
poster_path             243          2.43
release_date              0          0.00
title                     0          0.00
video                     0          0.00
vote_average              0          0.00
vote_count                0          0.00


## Resultados de la carga
- El dataset tiene 9,999 películas y 15 columnas
- Las columnas con más nulos son backdrop_path con 18.81% y overview con 2.78%
- El resto de columnas están completas sin nulos

## Limpieza de datos
Se aplica la limpieza del dataset: eliminación de columnas innecesarias,
conversión de tipos, manejo de nulos y normalización de géneros.

In [2]:
procesador = ProcesadorEda(df)
df_limpio = procesador.limpieza_datos()

Limpieza de datos completada
Columnas actuales: ['genre_ids', 'id', 'original_language', 'original_title', 'overview', 'popularity', 'release_date', 'title', 'vote_average', 'vote_count', 'anio_estreno']
Dataset limpio guardado en ../../data/processed/tmdb_movies_clean.csv


## Resultados de la limpieza

**Columnas eliminadas:**
- `Unnamed: 0` — índice sin valor informativo
- `adult` — todos los valores eran False, sin variación
- `video` — todos los valores eran False, sin variación
- `backdrop_path` — ruta de imagen, no útil para análisis (18.81% nulos)
- `poster_path` — ruta de imagen, no útil para análisis (2.43% nulos)

**Conversión de tipos:**
- `release_date` — convertido de string a datetime y se extrajo `anio_estreno`
- `genre_ids` — convertido de string a lista real con nombres de géneros

**Manejo de nulos:**
- `overview` — 278 nulos rellenados con "Sin descripción" ya que no es relevante para el EDA

**Dataset final:**
- 11 columnas y 9,999 filas
- Guardado en data/processed/tmdb_movies_clean.csv

## Resumen descriptivo
Estadísticas descriptivas de las variables numéricas principales del dataset.

In [3]:
procesador.resumen_descriptivo()

Resumen de descriptivo:
        popularity  vote_average    vote_count
count  9999.000000   9999.000000   9999.000000
mean      5.814750      4.551764    221.900790
std      13.101449      2.968713    879.465244
min       0.192900      0.000000      0.000000
25%       3.673650      0.000000      0.000000
50%       4.320600      5.708000      6.000000
75%       5.650550      6.724000     61.000000
max     951.346000     10.000000  21265.000000


,popularity,vote_average,vote_count
count,9999.000000,9999.000000,9999.000000
mean,5.814750,4.551764,221.900790
std,13.101449,2.968713,879.465244
min,0.192900,0.000000,0.000000
25%,3.673650,0.000000,0.000000
50%,4.320600,5.708000,6.000000
75%,5.650550,6.724000,61.000000
max,951.346000,10.000000,21265.000000


## Resultados del resumen descriptivo

**Popularidad:**
- Promedio de 5.8 pero máximo de 951, indica outliers significativos
- El 75% de las películas tiene popularidad menor a 5.65

**Calificación promedio (vote_average):**
- El 25% tiene calificación 0, correspondiente a películas sin votos aún
- La mediana real es 5.7 entre las películas calificadas

**Número de votos (vote_count):**
- La mitad de las películas tiene 6 votos o menos
- Máximo de 21,265 votos, hay películas muy votadas y otras casi sin votos
- Para análisis de calificaciones se filtra vote_count >= 50 para mayor confiabilidad